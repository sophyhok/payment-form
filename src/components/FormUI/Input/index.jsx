import React, { useState, useEffect } from 'react';
import { string, func, oneOf } from 'prop-types';
import classnames from 'classnames';

import styles from './input.module.scss';

const propTypes = {
  name: string,
  type: string,
  value: string,
  label: string,
  placeholder: string,
  onChange: func,
  boxStatus: oneOf(['none', 'success', 'danger']),
  testId: string,
}

const defaultProps = {
  name: 'input',
  type: 'text',
  value: '',
  placeholder: '',
  errorInput: '',
  onChange: () => { },
  boxStatus: 'none',
  testId: 'input'
}

const Input = ({
  name,
  type,
  value,
  placeholder,
  onChange,
  boxStatus,
  testId
}) => {
  const [inputValue, handleChange] = useState(value);

  const updateInput = event => {
    handleChange(event.target.value);
    onChange(event);
  };

  useEffect(() => {
    handleChange(value);
  }, [value]);

  return (
      <input
        className={classnames(styles.input, {[styles[`input__boxStatus_${boxStatus}`]]: boxStatus})}
        name={name}
        type={type}
        value={inputValue}
        placeholder={placeholder}
        onChange={event => updateInput(event)}
        data-testid={testId}
        pattern="[A-Za-z]{3}"
      />
  )
}

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default Input;