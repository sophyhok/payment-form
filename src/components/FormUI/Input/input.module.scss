.input {
  display: inline-flex;
  width: auto;
  background-color: #FFFFFF;
  height: 32px;
  padding: 0 16px;
  border: 1px solid #939FB1;
  border-radius: 5px;
  box-sizing: border-box;
  
  &__boxStatus_danger {
    border: 1px solid #FF0000;
  }

  &__boxStatus_success {
    border: 1px solid #00FF00;
  }

  &__boxStatus_none {
    border: 1px solid #939FB1;
  }
}
