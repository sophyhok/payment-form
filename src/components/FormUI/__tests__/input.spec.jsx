import React from 'react';
import { render } from '@testing-library/react';

import Input from '../Input';

it('should match snapshot', () => {
  const { asFragment} = render(<Input />);
  expect(asFragment()).toMatchSnapshot();
})

it('should return undefined when default onChange runs', () => {
  const result = Input.defaultProps.onChange();
  expect(result).toBe(undefined);
});

it('should render boxStatus success when props passed in', () => {
  const { getByTestId } = render(<Input boxStatus="success" />);
  const input = getByTestId('input');
  expect(input.className).toMatch('input__boxStatus_success');
});