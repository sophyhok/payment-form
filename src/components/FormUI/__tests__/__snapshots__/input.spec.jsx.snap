// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="input input__boxStatus_none"
    data-testid="input"
    name="input"
    pattern="[A-Za-z]{3}"
    placeholder=""
    type="text"
    value=""
  />
</DocumentFragment>
`;
